src/DynamoUtilities/DynamoPathManager.cs

```diff
@@ -364,10 +384,13 @@ public static bool PreloadAsmVersion(string version, DynamoPathManager pathManag
             Debug.WriteLine(string.Format("Attempting to preload ASM version {0}", version));
 
             string hostLocation;
-            if (!FindAsm(version, out hostLocation))
+            if (!FindAlternativeAsm(version, pathManager, out hostLocation))
```

Comment:
It would be good if the host application can set the ASM path, otherwise FindAsm always looks at "C:\Program Files\Autodesk\" whereas host application such as Revit might be installed in D drive or some other location. So in general host application knows where to load ASM from.